src/sqlfluff/core/parser/markers.py

```diff
@@ -3,16 +3,64 @@
 This class is a construct to keep track of positions within a file.
 """
 
-from collections import namedtuple
+from typing import Optional
 
-protoFilePositionMarker = namedtuple(
-    "FilePositionMarker", ["statement_index", "line_no", "line_pos", "char_pos"]
-)
 
-
-class FilePositionMarker(protoFilePositionMarker):
+class FilePositionMarker:
     """This class is a construct to keep track of positions within a file."""
 
+    slots = ["statement_index", "line_no", "line_pos", "char_pos"]
+
+    def __init__(
+        self,
+        statement_index: Optional[int],
+        line_no: int,
+        line_pos: Optional[int],
+        char_pos: int,
+    ):
+        # NB: statment_index and line_pos are optional so that
+        # we can use position markers to flag up errors which
+        # are specific to a line but we don't (or can't) know
+        # precisely where within that line the issue is.
+        self.statement_index = statement_index
+        self.line_no = line_no
+        self.line_pos = line_pos
+        self.char_pos = char_pos
+
+    def __str__(self):
+        return "[{0}]({1}, {2}, {3})".format(
+            self.char_pos, self.statement_index, self.line_no, self.line_pos
+        )
+
+    def __gt__(self, other):
+        return self.char_pos > other.char_pos
+
+    def __lt__(self, other):
+        return self.char_pos < other.char_pos
+
+    def __ge__(self, other):
+        return self.char_pos >= other.char_pos
+
+    def __le__(self, other):
+        return self.char_pos <= other.char_pos
+
+    def __eq__(self, other):
+        return (
+            (self.statement_index == other.statement_index)
+            and (self.line_no == other.line_no)
+            and (self.line_pos == other.line_pos)
+            and (self.char_pos == other.char_pos)
+        )
+
+    @classmethod
+    def from_fresh(cls):
+        """Construct a fresh position marker.
+
+        This should be the default way of creating a new position marker, and
+        is where we define what the `start` of a file looks like.
+        """
+        return cls(1, 1, 1, 0)
```

Comment:
Should these be defaults in `__init__` instead? and then the method can be removed

Reply:
🤦 - Good call. This is some *VERY* old code, which I've not really revisited in a while. 👍